src/XR/native/nativeXRFrame.ts

```diff
@@ -0,0 +1,98 @@
+import { NativeXRPlugin } from "./nativeXRPlugin";
+
+/** @hidden */
+interface INativeXRFrame extends XRFrame {
+    // Native supported functions
+    createAnchor(pose: XRRigidTransform, space: XRSpace): Promise<XRAnchor>;
+    fillPoses(spaces: XRSpace[], baseSpace: XRSpace, transforms: Float32Array): boolean;
+    getJointPose(joint: XRJointSpace, baseSpace: XRSpace): XRJointPose;
+    fillJointRadii(jointSpaces: XRJointSpace[], radii: Float32Array): boolean;
+
+    // Native-only helper functions
+    getPoseData: (space: XRSpace, baseSpace: XRReferenceSpace, vectorBuffer: ArrayBuffer, matrixBuffer: ArrayBuffer) => XRPose;
+}
+
+/** @hidden */
+export class NativeXRFrame implements XRFrame {
+    private _xrTransform = new XRRigidTransform();
+    private _xrPose: XRPose = {
+        transform: this._xrTransform,
+        emulatedPosition: false
+    };
+    // Enough space for position, orientation
+    private _xrPoseVectorData = new Float32Array(4 * (4 + 4));
+
+    public get session(): XRSession {
+        return this._nativeImpl.session;
+    }
+
+    constructor(private _nativeImpl: INativeXRFrame) { }
+
+    public getPose(space: XRSpace, baseSpace: XRReferenceSpace): XRPose | undefined {
+        if (!this._nativeImpl.getPoseData(space, baseSpace, this._xrPoseVectorData.buffer, this._xrTransform.matrix.buffer)) {

```

Comment:
Do these args need to be passed every time? Aren't they usually the same (especially `_xrPoseVectorData.buffer` and `_xrTransform.matrix.buffer`) every frame?

Reply:
This is a good point, although receiving the parameters in native doesn't seem to be significantly performance-heavy, especially when compared to pulling out the `Napi::External<xr::Space>` on the native side, so until we get that part much faster we won't see a difference if we ditch the extra parameters.